ravenframework/OutStreams/PlotInterfaces/GeneralPlot.py

```diff
@@ -136,204 +95,157 @@ def __init__(self):
+    self.dim = None          # default plot is 2D
+    self.sourceName = []     # list of source names
+    self.sourceData = None   # source of data
+    self.outStreamTypes = [] # list of the outstream types
+    self.destinations = None # where plots should go (screen, file, etc.)
+
+    # plot types key is data dimension, value is list of available plot types
+    self.availableOutStreamTypes = {2: ['scatter',
+                                        'line',
+                                        'histogram',
+                                        'stem',
+                                        'step',
+                                        'pseudocolor',
+                                        'dataMining',
+                                        'contour',
+                                        'filledContour'],
+                                    3: ['scatter',
+                                        'line',
+                                        'histogram',
+                                        'stem',
+                                        'surface',
+                                        'wireframe',
+                                        'tri-surface',
+                                        'contour',
+                                        'filledContour']}
+
+    # interpolators that can be used in plots
+    self.availableInterpolators = ['nearest',
+                                   'linear',
+                                   'cubic',
+                                   'multiquadric',
+                                   'inverse',
+                                   'gaussian',
+                                   'Rbflinear',
+                                   'Rbfcubic',
+                                   'quintic',
+                                   'thin_plate']
+
+    # plot components
+    self.fig = None               # figure
+    self.ax = None                # axes
+    self.actPlot = None           # plot action, ie., ax.plot()
+    self.gridSpace = None         # subplot setup
+    self.actcm = None             # colormap
+    self.xCoordinates = None      # x coordinate name
+    self.yCoordinates = None      # y coordinate name
+    self.zCoordinates = None      # z coordinate name
+    self.xValues = None           # dictionary of x values
+    self.yValues = None           # dictionary of y values
+    self.zValues = None           # dictionary of z values
+    self.colorMapCoordinates = {} # color map coordinates
+    self.colorMapValues = {}      # color map values
+
+    # For the data-mining plot, I think?
     self.clusterLabels = None
     self.clusterValues = None
 
-    ## Gaussian Mixtures
+    # Gaussian Mixtures
     self.mixtureLabels = None
     self.mixtureValues = None
     self.mixtureMeans = None
     self.mixtureCovars = None
 
-  # TODO started, but didn't finish due to time constraints
```

Comment:
this block absolutely should not be removed. It took me several days to get this far, and would be a shame to start over from scratch when this is inevitable fixed

Reply:
Okay, I'll put it back.